funnel/views/decorators.py

```diff
@@ -0,0 +1,17 @@
+from functools import wraps
+from flask import g, request, redirect, current_app
+from funnel import app, funnelapp
+
+
+def legacy_redirect(f):
+    @wraps(f)
+    def decorated_function(clview, **kwargs):
+        if g.profile:
+            if g.profile.legacy and current_app == app:
+                with funnelapp.app_context(), funnelapp.test_request_context():
```

Comment:
`test_request_context` should not be necessary.

Reply:
url_for doesn't work with only app_context. It needs request context. It's kind of a hack. Not an official feature of flask it seems. https://github.com/pallets/flask/issues/1412